### PR TITLE
Small improvements of java8 grammar

### PR DESCRIPTION
**Pull request by maybeec**

### CODE REVIEW

File: java8/Java8.g4 [hunk 1]
```diff
@@ -348,7 +348,8 @@ fieldModifier
 	;
 
 variableDeclaratorList
-	:	variableDeclarator (',' variableDeclarator)*
+	:	variableDeclarator
+	|	variableDeclaratorList ',' variableDeclarator
```
Comment (KvanTTT) [inline java8/Java8.g4]: What is the point of this change? I like the previous format.

Comment (maybeec) [inline java8/Java8.g4]: I thought it comes with better parse performance when turning it into left recursion. Just referring to one of your posts: https://github.com/antlr/grammars-v4/issues/528#issuecomment-304441606

Comment (KvanTTT) [inline java8/Java8.g4]: In this case, your rule has a worse performance. It does not reduce the number of rules in comparison of replacing a sequence of rules with a single left-recursive rule.

Comment (maybeec) [inline java8/Java8.g4]: Ok. So one essential metric for ANTLR performance is rule count?

File: java8/Java8.g4 [hunk 2]
```diff
@@ -705,7 +705,8 @@ arrayInitializer
 	;
 
 variableInitializerList
-	:	variableInitializer (',' variableInitializer)*
+	:	variableInitializer 
+	|	variableInitializerList ',' variableInitializer
```
Comment (maybeec) [inline java8/Java8.g4]: so this would be the same right?

Comment (KvanTTT) [inline java8/Java8.g4]: Correct.

File: java8/Java8.g4 (hunk 1 above)
Comment (KvanTTT) [inline java8/Java8.g4]: I adhere the following priority:
1. Rule count.
2. If possible replace left-recursive rule with non-left-recursive or no-recursion rule (your case).

Comment (maybeec) [inline java8/Java8.g4]: @KvanTTT How do fit no. of ATN states into the picture? I implemented a grammar transformation and had to make the decision whether I introduce new ATN states or new rules by alternatives.
As DFAs and NFAs are build up on demand during parsing, I was not aware of any other metric to take into account besides the ATN, which somehow indicates the language complexity.
So in my case also introducing new ATN states would have made more sense, right?

**Comment by teverett** [comment]
@KvanTTT Am i ok to pull this?

**Comment by KvanTTT** [comment]
Single-line change, but ok :)

File: java8/Java8.g4 (hunk 1 above)
Comment (KvanTTT) [inline java8/Java8.g4]: I don't understand clearly, but I think the number of ATN states depend on rule number. You can do your own research and compare performance results.

**Pull request merged**